### DISCUSSION TITLE
Add term to glossary is not shown in Glossary block in Translation UI

### DISCUSSION THREAD

**Discussion by Ere-Eye** [WeblateOrg/weblate#11520] [Q&A]
![image](https://github.com/WeblateOrg/weblate/assets/14802003/61525a59-7719-423d-8c82-fa1424097854)
I know that there is supposed to be a button at the bottom of the Glossary block that says "Add term to glossary", however, in my lastest project it is not visible. Previously, the Glossary components had alerts that it was complaining about so I completely revamped the component and it's clear of any alerts now. I also made sure "Manage strings" is checked in the translation settings but still unable to figure out why it doesn't show in the translation UI.

**Comment by Ere-Eye** [comment]
After some more tinkering, I found that the "Add term to glossary" is available when editing a component's source language. Which does not make much sense to me as I would hardly ever browse into translate the source strings (there wouldn't be anything to translate, such is shown in the UI as well.) It would make sense as a workflow to be able to add a glossary term on-the-go as one is translating from one language to another (e.g Chinese to English), and add that term's corresponding term in the language currently being translated. Does this merit creating an Issue from this discussion?

![image](https://github.com/WeblateOrg/weblate/assets/14802003/c8cb452d-3aff-441c-9b33-76973d053684)

![image](https://github.com/WeblateOrg/weblate/assets/14802003/024d5b75-53b0-4cc7-94c6-61f0d35747c4)

**Comment by nijel** [comment]
The button is there if you have permissions to add a term to the glossary.

**Reply by Ere-Eye** [reply]
I am the admin.

**Reply by nijel** [reply]
How is your glossary component configured?

**Reply by Ere-Eye** [reply]
File format: CSV file
☑ Use as a glossary
☑ Manage strings
☑ Edit base file

Did you by any chance see the reply added under this thread? When I browse the Glossary's source language, "Add term to glossary" is visible.

**Reply by nijel** [reply]
Do you have a monolingual base set? Monolingual setup defines strings by the base language, so you cannot add strings to translations only.

PS: https://github.com/WeblateOrg/weblate/pull/11590 will add an alert in that case.

**Reply by Ere-Eye** [reply]
Yes, as I have glossary terms translated into multiple languages that I upload during setup of the project. Though I don't see how that is in conflict with adding glossary terms while translating. I don't want to add strings only to the translation as that wouldn't do me any good. I want to take a term from the source I translated (say a fictional place name) and add that in the glossary with its corresponding translation to ensure consistency as you go further in the translation process. Like this:
![image](https://github.com/WeblateOrg/weblate/assets/14802003/417a728d-6a37-4966-8eec-eee96e7f74f7)
Then the other languages will have a new string available for translation too.
